core/FrontController.php

```diff
@@ -493,7 +493,8 @@ protected function prepareDispatch($module, $action, $parameters)
 
     protected function handleMaintenanceMode()
     {
-        if ((Config::getInstance()->General['maintenance_mode'] != 1) || Common::isPhpCliMode()) {
+        // as request matomo behind load balancer should not return 503. https://github.com/matomo-org/matomo/issues/18054
+        if ((Config::getInstance()->General['maintenance_mode'] != 1) || Common::isPhpCliMode() || Config::getInstance()->General['multi_server_environment']) {
```

Comment:
I don't think it's on purpose to disable the maintenance mode on multi server environments, is it?
I actually understood the issue in a way, that only the response code should be changed to `200` in that case...